neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -314,6 +347,28 @@ private void OnVersionMessageReceived(VersionPayload payload)
             Context.Parent.Tell(payload);
         }
 
+        private void OnTimer()
+        {
+            RefreshPendingKnownHashes();
+        }
+
+        protected override void PostStop()
+        {
+            timer.CancelIfNotNull();
+            base.PostStop();
+        }
+
+        private void RefreshPendingKnownHashes()
+        {
+            while (pendingKnownHashes.Count > 0)
+            {
+                KeyValuePair<UInt256, DateTime> item = pendingKnownHashes.First();
+                if (DateTime.UtcNow - item.Value <= PendingTimeout)
```

Comment:
Use `TimeProvider`

Reply:
Think the code is from the existing project, I did want to change it, but I found many places are using `DateTime.UtcNow`, so I keep it as the original version.